lib/webidl2.js

```diff
@@ -454,16 +454,16 @@
     }
 
     function type_with_extended_attributes(typeName) {
-      const extAttrs = extended_attrs();
+      const extAttrs = extended_attrs() || null;
```

Comment:
I wonder if extended_attrs() should just return null when appropriate?

Reply:
I felt I shouldn't do that as no other functions return null, but yeah, I think that's better.